lib/_http_client.js

```diff
@@ -447,16 +447,26 @@ function socketOnData(d) {
       socket.destroy();
     }
   } else if (parser.incoming && parser.incoming.complete &&
-             // When the status code is 100 (Continue), the server will
-             // send a final response after this client sends a request
-             // body. So, we must not free the parser.
-             parser.incoming.statusCode !== 100) {
+             // When the status code is informational (100, 102-199),
+             // the server will send a final response after this client
+             // sends a request body, so we must not free the parser.
+             // 101 (Switching Protocols) and all other status codes
+             // should be processed normally.
+             !statusIsInformational(parser.incoming.statusCode)) {
     socket.removeListener('data', socketOnData);
     socket.removeListener('end', socketOnEnd);
     freeParser(parser, req, socket);
   }
 }
 
+// client
```

Comment:
I am not sure what this comment stands for.

Reply:
I do not either. I was following what looked like existing convention in this source file. Removing.